### PR TITLE
API Gateway feature

### PR DESCRIPTION
**Pull request by petergrlica**
Added api gateway to server.

Please check mostly pkg/apigw, as the system CRUD is fairly common.

### CODE REVIEW

File: automation/automation/jsenv_handler.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    meta:
+      short: Process arbitrary data in jsenv
+    params:
+      scope:
+        required: true
+        types:
+          - { wf: Any }
+          - { wf: String, suffix: String }
+          - { wf: Reader, suffix: Stream }
+      function:
```
Comment (darh) [inline automation/automation/jsenv_handler.yaml]: would "source" be a better name for this here?

File: automation/automation/jsenv_handler.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+imports:
+  - io
+
+functions:
+  process:
```
Comment (darh) [inline automation/automation/jsenv_handler.yaml]: maybe `execute` instead of process?

File: automation/automation/jsenv_handler.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	out, err := fn.Exec(vm.New(args.scopeString))
+
+	if err != nil {
+		err = fmt.Errorf("could not exec jsenv function: %s", err)
+		return
+	}
+
+	switch vv := out.(type) {
+	case uint64, int64:
+		res.ResultInt = int64(vv.(uint64))
```
Comment (darh) [inline automation/automation/jsenv_handler.go:61]: can you just add a quick comment to these two that they might be removed (if I understand you correctly, this is the case here, right?)

**Review by darh — COMMENTED** [review]
non obligatory suggestions for improvements

File: automation/automation/jsenv_handler.yaml (hunk 1 above)
Comment (petergrlica) [reply automation/automation/jsenv_handler.yaml]: it would, thanks

File: automation/automation/jsenv_handler.go (hunk 3 above)
Comment (petergrlica) [reply automation/automation/jsenv_handler.go:61]: :+1:

**Pull request merged**